Plugin Sql Tests for WPOrgPluginModel & PluginDirectoryModel

oguzkocer opened the pull request:
This PR adds some unit tests for `PluginSqlUtils` methods. While working on this PR, I realized that we are using a class called `SingleStoreWellSqlConfigForTests` which setups the tests for a single type of model. Since `PluginSqlUtils` deals with 3 different models, I was unable to add these tests to the same class and had to separate each one. Due to this limitation, the only method I couldn't test that I wanted to was `getWPOrgPluginsForDirectory`. We'd need a different setup to make this work, but that's not something I am looking to do right now. Aside from that the PR is pretty straightforward to review & test.

This is the last PR before we merge `feature/plugin-directory-master` to `develop`. I'll open that PR as soon as this is merged. The remaining minor issues about plugins will be handled separately.

aforcier commented:
@oguzkocer you should be able to support multiple models in one test if you use the usual `new WellSqlConfig(context)` call and skip `SingleStoreWellSqlConfigForTests`, IIRC. [`UploadStoreUnitTest`](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/54c164ccd8ee723fe74e297c0ac0736ce0f8f189/example/src/test/java/org/wordpress/android/fluxc/upload/UploadStoreUnitTest.java) does this, for example.

oguzkocer commented:
@aforcier I did not know that, thanks a lot for pointing it out!

@nbradbury In that case, I would like to move the tests to one class and add that missing test. Do you mind holding off on the review until then? I'll ping you once it's ready.

oguzkocer commented:
@nbradbury I've decided to keep the 3 separate test classes, because adding all of them into a single file made it really hard to navigate the code. As a result, I only needed to change the `PluginDirectorySqlUtilsTest`'s setup and added the test for `getWPOrgPluginsForDirectory` there. Unfortunately that test is a little complicated, but it helped me fix a minor issue with a query in `insertOrUpdatePluginDirectoryModel`.

I've also added reflection to assert a few more things in the tests after opening the PR. Hope that's OK!

It should be ready to review now! 🤞

nbradbury commented on example/src/test/java/org/wordpress/android/fluxc/plugin/PluginDirectorySqlUtilsTest.java:
```diff
@@ -0,0 +1,191 @@
+
+@RunWith(RobolectricTestRunner.class)
+public class PluginDirectorySqlUtilsTest {
+    private Random mRandom = new Random(System.currentTimeMillis());
```
Minor nit: `mRandom` can be `final` in all three of these classes.

nbradbury commented:
@oguzkocer Code looks clean and all tests pass. I'm good to :shipit: despite the minor nit about `mRandom` not being final (that can be done another time, if necessary).
